### PR TITLE
[ELF] Resolve .eh_frame symbols renamed by GCC LTO.

### PR DESCRIPTION
**Pull request by ishitatsuyuki**
Closes: #977.

See commit descriptions.

### CODE REVIEW

File: elf/input-files.cc [hunk 1]
```diff
@@ -324,7 +324,6 @@ void ObjectFile<E>::initialize_ehframe_sections(Context<E> &ctx) {
     std::unique_ptr<InputSection<E>> &isec = sections[i];
     if (isec && isec->is_alive && isec->name() == ".eh_frame") {
       read_ehframe(ctx, *isec);
-      isec->is_alive = false;
```
Comment (rui314) [inline elf/input-files.cc:327]: If a section is not copied from input files to an output file as-is, I believe we mark it as dead. So we want to keep this code.

**Comment by dhewg** [comment]
This fixes the build error from the linked issue for me, thanks!
Not runtime tested yet though

**Comment by ishitatsuyuki** [comment]
(Currently a bit busy with other stuff, will try to address the review comment soon)

File: elf/input-files.cc (hunk 1 above)
Comment (ishitatsuyuki) [reply elf/input-files.cc:327]: I switched to an approach that is similar to what is used for mergeable sections, i.e. killing the sections after symbol resolution.

**Comment by ishitatsuyuki** [comment]
@dhewg Would appreciate retesting the new version (and also testing the linked binary if possible).

**Comment by dhewg** [comment]
I gave it a quick spin, and it builds and works just fine!
Thanks!

**Comment by ishitatsuyuki** [comment]
Thanks for testing. I've moved the PR out of draft.

**Pull request merged**